test/parallel/test-stream-duplex-readable-end.js

```diff
@@ -0,0 +1,30 @@
+'use strict';
+// https://github.com/nodejs/node/issues/35926
+const common = require('../common');
+const stream = require('stream');
+
+let loops = 3;
+
+const src = new stream.Readable({
+  read: common.mustCallAtMost(() => {
+    if (loops--)
+      src.push(Buffer.alloc(20000));
+  }, 2)
+});
+
+const dst = new stream.Transform({
+  transform(chunk, output, fn) {
+    this.push(null);
+    fn();
+  }
+});
+
+src.pipe(dst);
+
+function parser_end(error) {
+  dst.removeAllListeners();
+}
+
+dst.on('data', data => console.log(data));
```

Comment:
please remove the `console.log` and use a `common.mustCall()` with an assertion instead.

Reply:
I reshuffled it a little bit, `'data'` doesn't get called on correct execution - but must be there in order to trigger the reading